Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -1281,6 +1288,11 @@ static String mutationObserverNotificationString()
 void UnifiedPDFPlugin::didBeginMagnificationGesture()
 {
     m_inMagnificationGesture = true;
+
+#if ENABLE(UNIFIED_PDF_DATA_DETECTION)
+    hideActiveDataDetectorHighlightOverlay();
+    dataDetectorOverlayController().setCanPresentHighlightOverlays(false);
```

Comment:
Rather than setBlah(false) and setBlah(true), can't dataDetectorOverlayController call back to a function on the plugin like `isOKToShowHightlight() const { return ! m_inMagnificationGesture; }` ?

Reply:
Good point, thanks. This reduces how much state is floating around.

Reply:
Fixed in https://github.com/WebKit/WebKit/commit/5bd05389d6e9401e0d119fc1afe258157c7c136f